ports/espressif/boards/sunton_esp32_8048S070/pins.c

```diff
@@ -0,0 +1,125 @@
+        MP_ROM_PTR(&pin_GPIO5),
+        MP_ROM_PTR(&pin_GPIO4),
+    }
+};
+
+STATIC const mp_rom_map_elem_t tft_pins_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_de), MP_ROM_PTR(&pin_GPIO41) },
+    { MP_ROM_QSTR(MP_QSTR_vsync), MP_ROM_PTR(&pin_GPIO40) },
+    { MP_ROM_QSTR(MP_QSTR_hsync), MP_ROM_PTR(&pin_GPIO39) },
+    { MP_ROM_QSTR(MP_QSTR_dclk), MP_ROM_PTR(&pin_GPIO42) },
+    { MP_ROM_QSTR(MP_QSTR_red), MP_ROM_PTR(&tft_r_pins) },
+    { MP_ROM_QSTR(MP_QSTR_green), MP_ROM_PTR(&tft_g_pins) },
+    { MP_ROM_QSTR(MP_QSTR_blue), MP_ROM_PTR(&tft_b_pins) },
+};
+MP_DEFINE_CONST_DICT(tft_pins_dict, tft_pins_table);
+
+STATIC const mp_rom_map_elem_t timings_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_frequency), MP_ROM_INT(12500000) },
+    { MP_ROM_QSTR(MP_QSTR_width), MP_ROM_INT(800) },
+    { MP_ROM_QSTR(MP_QSTR_height), MP_ROM_INT(480) },
+    { MP_ROM_QSTR(MP_QSTR_hsync_pulse_width), MP_ROM_INT(30) },
+    { MP_ROM_QSTR(MP_QSTR_hsync_back_porch), MP_ROM_INT(16) },
+    { MP_ROM_QSTR(MP_QSTR_hsync_front_porch), MP_ROM_INT(210) },
+    { MP_ROM_QSTR(MP_QSTR_hsync_idle_low), MP_ROM_TRUE },
+    { MP_ROM_QSTR(MP_QSTR_vsync_pulse_width), MP_ROM_INT(13) },
+    { MP_ROM_QSTR(MP_QSTR_vsync_back_porch), MP_ROM_INT(10) },
+    { MP_ROM_QSTR(MP_QSTR_vsync_front_porch), MP_ROM_INT(22) },
+    { MP_ROM_QSTR(MP_QSTR_vsync_idle_low), MP_ROM_TRUE },
+    { MP_ROM_QSTR(MP_QSTR_de_idle_high), MP_ROM_FALSE },
+    { MP_ROM_QSTR(MP_QSTR_pclk_active_high), MP_ROM_FALSE },
+    { MP_ROM_QSTR(MP_QSTR_pclk_idle_high), MP_ROM_FALSE },
+
+};
+MP_DEFINE_CONST_DICT(timings_dict, timings_table);
+
+STATIC const mp_rom_map_elem_t board_module_globals_table[] = {
+    CIRCUITPYTHON_BOARD_DICT_STANDARD_ITEMS
+
+    // Display constructs
+    { MP_ROM_QSTR(MP_QSTR_TFT_PINS), MP_ROM_PTR(&tft_pins_dict) },
+    { MP_ROM_QSTR(MP_QSTR_TFT_TIMINGS), MP_ROM_PTR(&timings_dict) },
+    { MP_ROM_QSTR(MP_QSTR_TFT_BACKLIGHT), MP_ROM_PTR(&pin_GPIO2) },
+
+    // User buttons
+    { MP_ROM_QSTR(MP_QSTR_IO0), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_BOOT0), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_BUTTON), MP_ROM_PTR(&pin_GPIO0) },
+
+    // User accessible GPIO
+    { MP_ROM_QSTR(MP_QSTR_IO19), MP_ROM_PTR(&pin_GPIO19) }, // P2, External SPI plug
+    { MP_ROM_QSTR(MP_QSTR_IO11), MP_ROM_PTR(&pin_GPIO11) },
+    { MP_ROM_QSTR(MP_QSTR_IO12), MP_ROM_PTR(&pin_GPIO12) },
+    { MP_ROM_QSTR(MP_QSTR_IO13), MP_ROM_PTR(&pin_GPIO13) },
+
+    { MP_ROM_QSTR(MP_QSTR_IO20), MP_ROM_PTR(&pin_GPIO20) }, // P3 has 19&20 for I2C bus, plus 17&18
+    { MP_ROM_QSTR(MP_QSTR_IO17), MP_ROM_PTR(&pin_GPIO17) }, // P4 & P5 are both 17,18,3.3v,G
+    { MP_ROM_QSTR(MP_QSTR_IO18), MP_ROM_PTR(&pin_GPIO18) },
+
+    // UART pins
+    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_GPIO43) },
+    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_GPIO44) },
+
+    // I2C
+    { MP_ROM_QSTR(MP_QSTR_SDA), MP_ROM_PTR(&pin_GPIO19) },
+    { MP_ROM_QSTR(MP_QSTR_SCL), MP_ROM_PTR(&pin_GPIO20) },
```

Comment:
I believe GPIO 18 and 19 are used by ESPRESSIF chips for USB D+/D- and CircuitPython doesn't share them very well (you can use them with Arduino code). I believe the Resistive Touch version of this board will have problems using the Touch Screen in CircuitPython. Do you think 17 & 18 would be a better choice for the board's generic I2C bus and pins 19 & 20 would be better named TOUCH_SCL & TOUCH_SDA? I know the touch I2C pins are broken out on P3 but given the USB conflict and that P4 and P5 look to me like they're designed for I2C bus usage as well they might be work out better.

Reply:
I have added definitions for those pins for i2s usage.  It doesn't appear as if the i2c touch sensor (the capacitive one, which I have) has any issue with utilizing pins 19/20.  I agree that it looks like 17/18 would be good candidates for i2c, but attempting to use them results in an in use error, I'm guessing due to the i2s amplifier keeping them active.

Reply:
ugh..  I've messed up the PR..

Reply:
Yea the USB conflict probably only affects the resisitve touch screen. The I2S conflict is a good guess for the the "use error" but I'd also be surprised if attempting to use the I2C bus over pins 19&20 worked, I'd expect some sort of resource in use error from that as well.

Reply:
the resistive is spi, the capacitive (the one I have) uses i2c on 19/20 and it works as far as I can tell.  touches register, no errors..

Reply:
hmm, have you tested the touch detection in CircuitPython? I'm really shocked that it works.

Reply:
Yes.